src/compiler/scala/tools/nsc/transform/Erasure.scala

```diff
@@ -582,17 +584,16 @@ abstract class Erasure extends InfoTransform
       bridge setInfo (otpe cloneInfo bridge)
       bridgeTarget(bridge) = member
 
-      def sigContainsValueClass = (member.tpe exists (_.typeSymbol.isDerivedValueClass))
-
-      val shouldAdd = (
-            !sigContainsValueClass
-        ||  (checkBridgeOverrides(member, other, bridge) match {
-              case Nil => true
-              case es if member.owner.isAnonymousClass => resolveAnonymousBridgeClash(member, bridge); true
-              case es => for ((pos, msg) <- es) reporter.error(pos, msg); false
-            })
-      )
-
+      val shouldAdd = {
+        val sigContainsValueClass = member.tpe.exists(_.typeSymbol.isDerivedValueClass)
+        def bridgeMayClash = bridge.paramss.exists(_.exists(p => p.tpe =:= ObjectTpe || isTrivialTopType(p.tpe)))
+        def bridgeIsAOK = checkBridgeOverrides(member, other, bridge) match {
+          case Nil => true
+          case es if member.owner.isAnonymousClass => resolveAnonymousBridgeClash(member, bridge); true
+          case es => for ((pos, msg) <- es) reporter.error(pos, msg); false
+        }
+        if (sigContainsValueClass || bridgeMayClash) bridgeIsAOK else true
```

Comment:
perhaps a little clearer as:

```scala
!(sigContainsValueClass || bridgeMayClash) || bridgeIsAOK
```

or not, matter of taste I guess?

Reply:
It took me two tries to get it right, so I decided not to be clever, but I am willing to upgrade.